dunwoody_disaster/views/MonologueWidget.py

```diff
@@ -0,0 +1,152 @@
+import pygame
+import json
+from PySide6.QtWidgets import (
+    QWidget,
+    QLabel,
+    QTextEdit,
+    QPushButton,
+    QVBoxLayout,
+    QHBoxLayout,
+    QSizePolicy,
+)
+from PySide6.QtGui import QPixmap, QFont, QKeyEvent
+from PySide6.QtCore import QTimer, Qt
+from typing import Callable
+
+from dunwoody_disaster import AUDIO
+import dunwoody_disaster as DD
+from dunwoody_disaster.CharacterFactory import Character
+
+
+# Define a QWidget-based class to manage monologue display in a game
+class MonologueWidget(QWidget):
+    # Initialize the widget with a character object and a transition callback function
+    def __init__(self, char: Character, transition_callback: Callable):
+        super().__init__()
+        self.char = (
+            char  # Store reference to the character whose dialogue will be displayed
+        )
+        self.current_dialogue_index = (
+            0  # Index to keep track of the current dialogue being displayed
+        )
+        self.char_index = 0  # Index to keep track of the current character being displayed in the typewriter effect
+        self.transition_callback = (
+            transition_callback  # Store the callback function for transition
+        )
+
+        self.initUI()  # Setup the user interface components
+        self.initSound()  # Setup sound components
+        self.loadDialogue()  # Load dialogue data from files
+
+    def initUI(self):
+        # Create the main layout
+        mainLayout = QHBoxLayout(self)
+
+        # Configures the character image, making it larger and positioned to the left
+        self.char_img = QLabel(self)
+        self.char_img.setPixmap(
+            QPixmap(self.char.image()).scaled(800, 800, Qt.KeepAspectRatio)
+        )  # Scale image
+        self.char_img.setSizePolicy(QSizePolicy.Preferred, QSizePolicy.Expanding)
+        mainLayout.addWidget(self.char_img)
+
+        # Create a layout for text and buttons on the right
+        textLayout = QVBoxLayout()
+
+        # Text edit to show dialogue
+        self.dialogueText = QTextEdit(self)
+        self.dialogueText.setFont(QFont("Arial", 14))
+        self.dialogueText.setReadOnly(True)
+        self.dialogueText.setMaximumHeight(800)  # Limit the height of the chatbox
+        textLayout.addWidget(self.dialogueText)
+
+        # Button to move to the next dialogue
+        self.nextButton = QPushButton("Next", self)
+        self.nextButton.clicked.connect(self.display_next_dialogue)
+        self.nextButton.setMaximumHeight(50)  # Limit the height of the button
+        textLayout.addWidget(self.nextButton)
+
+        # Add textLayout to the mainLayout
+        mainLayout.addLayout(textLayout, 1)
+
+        self.timer = QTimer(self)
+        self.timer.timeout.connect(
+            self.add_char
+        )  # Timer to control the typewriter effect
+
+    def initSound(self):
+        pygame.mixer.init()
```

Comment:
I wonder if we can make a audio manager in `__init__.py` because I believe it is only run once. From there it would be as simple as `DD.AudioManager.set_audio()` from anywhere. That way we can avoid initializing it so much.

Reply:
That would be great but I haven't looked into it since I just wanted to use what worked for now.